src/components/IGVFileSelector.js

```diff
@@ -13,27 +12,80 @@ const getStrings = (v) => {
   return Utils.cond([_.isString(v), () => [v]], [!!v?.items, () => _.flatMap(getStrings, v.items)], () => []);
 };
 
-export const getValidIgvFiles = (values) => {
-  return _.flatMap((value) => {
-    const possibleFile = /(.+)\.([^.]+)$/.exec(value);
+const splitExtension = (fileUrl) => {
+  const extensionDelimiterIndex = fileUrl.lastIndexOf('.');
+  const base = fileUrl.slice(0, extensionDelimiterIndex);
+  const extension = fileUrl.slice(extensionDelimiterIndex + 1);
+  return [base, extension];
+};
 
-    if (_.isEmpty(parseGsUri(value)) || !possibleFile) {
-      return [];
-    }
+const isTdrUrl = (fileUrl) => {
+  const parts = fileUrl.split('/');
+  return (
+    parts.length === 6 &&
+    /datarepo-[a-f0-9]+-bucket/.test(parts[2]) &&
+    /[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}/.test(parts[3]) &&
```

Comment:
I see this UUID regex used in a few places, it might be easier to grok and maintain if it's defined separately and reused.

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/4084/commits/a2f179f63aa4d951add20b3b4fc260e75d60563b.